src/libraries/System.Private.CoreLib/src/System/Buffers/Text/Ascii.cs

```diff
@@ -0,0 +1,1244 @@
+        /// One buffer is provided as 8-bit <see cref="byte"/>s. The other buffer is provided
+        /// as 16-bit <see cref="char"/>s.
+        /// </summary>
+        /// <param name="left">The first ASCII text buffer to compare.</param>
+        /// <param name="right">The second ASCII text buffer to compare.</param>
+        /// <returns><see langword="true"/> if the buffers are equivalent; otherwise, <see langword="false"/>.</returns>
+        public static bool Equals(ReadOnlySpan<byte> left, ReadOnlySpan<char> right)
+        {
+            if (left.Length != right.Length)
+            {
+                return false;
+            }
+
+            for (int i = 0; i < left.Length; i++)
+            {
+                byte leftValue = left[i];
+                if (!IsAscii(leftValue) || leftValue != right[i])
+                {
+                    return false;
+                }
+            }
+
+            return true;
+        }
+
+        /// <summary>
+        /// Returns a value stating whether the contents of two ASCII text buffers are equal.
+        /// One buffer is provided as 8-bit <see cref="byte"/>s. The other buffer is provided
+        /// as 16-bit <see cref="char"/>s.
+        /// </summary>
+        /// <param name="left">The first ASCII text buffer to compare.</param>
+        /// <param name="right">The second ASCII text buffer to compare.</param>
+        /// <returns><see langword="true"/> if the buffers are equivalent; otherwise, <see langword="false"/>.</returns>
+        /// <exception cref="ArgumentNullException"><paramref name="right"/> is <see langword="null"/>.</exception>
+        public static bool Equals(ReadOnlySpan<byte> left, string right)
+        {
+            if (right is null)
+            {
+                throw new ArgumentNullException(nameof(right));
+            }
+
+            return Equals(left, right.AsSpan());
+        }
+
+        /// <summary>
+        /// Returns a value stating whether the contents of two ASCII text buffers are equal
+        /// using a case-insensitive comparison. One buffer is provided as 8-bit <see cref="byte"/>s.
+        /// The other buffer is provided as 16-bit <see cref="char"/>s.
+        /// </summary>
+        /// <param name="left">The first ASCII text buffer to compare.</param>
+        /// <param name="right">The second ASCII text buffer to compare.</param>
+        /// <returns><see langword="true"/> if the buffers are equivalent; otherwise, <see langword="false"/>.</returns>
+        public static bool EqualsIgnoreCase(ReadOnlySpan<byte> left, ReadOnlySpan<char> right)
+        {
+            if (left.Length != right.Length)
+            {
+                return false;
+            }
+
+            for (int i = 0; i < left.Length; i++)
+            {
+                byte leftValue = left[i];
+                if (!IsAscii(leftValue) || ToLower(leftValue) != ToLower(right[i]))
```

Comment:
Why is it useful at https://github.com/dotnet/runtime/pull/35386/files#diff-ee7e74f268c753b1c4a62f442af7aaf6R103 to check the identical casing first but not here?